azurerm/resource_arm_autoscale_setting.go

```diff
@@ -0,0 +1,954 @@
+						},
+						"webhook": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							MinItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"service_uri": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"properties": {
+										Type:     schema.TypeMap,
+										Optional: true,
+									},
+								},
+							},
+							Set: resourceAzureRmAutoscaleDefaultHash,
+						},
+					},
+				},
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+			"target_resource_uri": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmAutoscaleSettingCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	armClient := meta.(*ArmClient)
+	asClient := armClient.autoscaleSettingsClient
+
+	name := d.Get("name").(string)
+	resourceGroupName := d.Get("resource_group_name").(string)
+	resourceType := "Microsoft.Insights/autoscaleSettings"
+	location := d.Get("location").(string)
+	enabled := d.Get("enabled").(bool)
+	targetResourceURI := d.Get("target_resource_uri").(string)
+	tags := d.Get("tags").(map[string]interface{})
+	expandedTags := expandTags(tags)
+
+	profiles, err := expandAzureRmAutoscaleProfile(d)
+	if err != nil {
+		return err
+	}
+
+	notifications, err := expandAzureRmAutoscaleNotification(d)
+	if err != nil {
+		return err
+	}
+
+	autoscaleSetting := insights.AutoscaleSetting{
+		Name:              &name,
+		Enabled:           &enabled,
+		TargetResourceURI: &targetResourceURI,
+		Profiles:          &profiles,
+		Notifications:     &notifications,
+	}
+
+	parameters := insights.AutoscaleSettingResource{
+		Name:             &name,
+		Type:             &resourceType,
+		Location:         &location,
+		Tags:             expandedTags,
+		AutoscaleSetting: &autoscaleSetting,
+	}
+
+	result, err := asClient.CreateOrUpdate(resourceGroupName, name, parameters)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(*result.ID)
+
+	return resourceArmAutoscaleSettingRead(d, meta)
+}
+
+func resourceArmAutoscaleSettingRead(d *schema.ResourceData, meta interface{}) error {
+	asClient := meta.(*ArmClient).autoscaleSettingsClient
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["autoscalesettings"]
+
+	if name == "" {
+		return fmt.Errorf("Cannot find resource name in Resource ID for Autoscaling Setting")
+	}
+
+	result, err := asClient.Get(resGroup, name)
+	if err != nil {
+		if result.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on Autoscaling Setting %s: %s", name, err)
+	}
+
+	autoscaleSetting := *result.AutoscaleSetting
+	d.Set("name", result.Name)
+	d.Set("resource_group_name", resGroup)
+	d.Set("location", result.Location)
+	d.Set("enabled", autoscaleSetting.Enabled)
```

Comment:
sometimes collections of properties, like `result.AutoscaleSetting` can come back nil, so we'd usually check for that